include/alpaka/workdiv/WorkDivHelpers.hpp

```diff
@@ -304,87 +303,103 @@ namespace alpaka
+        [[maybe_unused]] TDev const& dev,
+        TKernelFnObj const& kernelFnObj,
+        TArgs&&... args) -> WorkDivMembers<Dim<TAcc>, Idx<TAcc>>
+    {
+        using Acc = TAcc;
 
         // Get max number of threads per block depending on the kernel function attributes.
         // For GPU backend; number of registers used by the kernel, local and shared memory usage of the kernel
         // determines the max number of threads per block. This number could be equal or less than the max number of
         // threads per block defined by device properties.
-        auto const kernelFunctionAttributes = getFunctionAttributes<Acc>(dev, kernelBundle);
+        auto const kernelFunctionAttributes
+            = getFunctionAttributes<Acc>(dev, kernelFnObj, std::forward<TArgs>(args)...);
         auto const threadsPerBlock = kernelFunctionAttributes.maxThreadsPerBlock;
 
         if constexpr(Dim<TGridElemExtent>::value == 0)
         {
             auto const zero = Vec<DimInt<0>, Idx<Acc>>{};
-            ALPAKA_ASSERT(gridElemExtent == zero);
-            ALPAKA_ASSERT(threadElemExtents == zero);
+            ALPAKA_ASSERT(kernelCfg.gridElemExtent == zero);
+            ALPAKA_ASSERT(kernelCfg.threadElemExtents == zero);
             return WorkDivMembers<DimInt<0>, Idx<Acc>>{zero, zero, zero};
         }
         else
             return subDivideGridElems(
-                getExtents(gridElemExtent),
-                getExtents(threadElemExtents),
+                getExtents(kernelCfg.gridElemExtent),
+                getExtents(kernelCfg.threadElemExtents),
                 getAccDevProps<Acc>(dev),
                 static_cast<Idx<Acc>>(threadsPerBlock),
-                blockThreadMustDivideGridThreadExtent,
-                gridBlockExtentSubDivRestrictions);
+                kernelCfg.blockThreadMustDivideGridThreadExtent,
+                kernelCfg.gridBlockExtentSubDivRestrictions);
 
         using V [[maybe_unused]] = Vec<Dim<TGridElemExtent>, Idx<TGridElemExtent>>;
         ALPAKA_UNREACHABLE(WorkDivMembers<Dim<TGridElemExtent>, Idx<TGridElemExtent>>{V{}, V{}, V{}});
     }
 
+    //! Checks if the work division is supported
+    //!
+    //! \tparam TWorkDiv The type of the work division.
     //! \tparam TDim The dimensionality of the accelerator device properties.
     //! \tparam TIdx The idx type of the accelerator device properties.
-    //! \tparam TWorkDiv The type of the work division.
-    //! \param accDevProps The maxima for the work division.
     //! \param workDiv The work division to test for validity.
+    //! \param accDevProps The maxima for the work division.
     //! \return If the work division is valid for the given accelerator device properties.
-    template<typename TDim, typename TIdx, typename TWorkDiv>
-    ALPAKA_FN_HOST auto isValidWorkDiv(AccDevProps<TDim, TIdx> const& accDevProps, TWorkDiv const& workDiv) -> bool
+    template<, typename TWorkDiv, typename TDim, typename TIdx>
```

Comment:
```suggestion
    template<typename TWorkDiv, typename TDim, typename TIdx>
```
?

Reply:
type during the last refactroing of the template argument order :-(

Reply:
fixed